tasks/libs/pipeline/tools.py

```diff
@@ -19,6 +20,55 @@
 PIPELINE_FINISH_TIMEOUT_SEC = 3600 * 5
 
 
+# Reference: https://docs.gitlab.com/api/graphql/reference/#cijobstatus
+class GitlabJobStatus(Enum):
+    CANCELED = "canceled"  # A job that is canceled.
```

Comment:
```suggestion
    canceled = auto()
```
I think the comment is useless and you can get the string with `GitlabJobStatus.canceled.name`. But that might not simplify the code, you can probably ignore this one